src/praisonai/praisonai/__init__.py

```
@@ -217,8 +217,17 @@ def __getattr__(name):
 
     try:
         import praisonaiagents
-    except ModuleNotFoundError:
-        raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
+    except ModuleNotFoundError as e:
+        if e.name == "praisonaiagents":
+            # Genuinely not installed — this is a real attribute miss.
+            raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)